server/src/modules/users/infra/repos/implementations/mikro-user-repo.ts

```diff
@@ -12,36 +15,52 @@ export class MikroUserRepo implements UserRepo {
   // right now we are directly importing the DB via module which is bad bc it couples this repo to the mikroentityrepo which is not good bc of testing. ideally we would pass this repo an entityrepostub
 
   // look into async imports with JS (one option i tried exploring is inverisfy, that shit is confusing though with async modules)
-  private usersEntityRepo: EntityRepository<UserEntity> | undefined
-  constructor(userRepo: EntityRepository<UserEntity> | undefined) {
-    this.usersEntityRepo = userRepo
+  constructor(_userRepo: EntityRepository<UserEntity> | undefined) {
+    
   }
 
-  async exists(userEmail: UserEmail): Promise<boolean> {
+  async exists(userEmail: UserEmail): Promise<Result<boolean, DBErrors>> {
     const user = await DB.usersEntityRepo.findOne({ email: userEmail.value })
-    return user !== null
+    return Result.ok(user !== null)
```

Comment:
Does moving control flow outside read nicer?

`if (user) { return Result.ok(true) } else { ...}`
or
`return user !== null ? Result.ok(true) : Result.ok(false)`

Reply:
Yeh I agree, much more readable (was kinda impatient when coding that aha)